test/dynamo/test_misc.py

```diff
@@ -11371,6 +11371,117 @@ def get_rng():
         self.assertEqual(expected, actual)
         self.assertGreater(po.call_count, 0)
 
+    def test_graph_output_count(self):
+        test_case_self = self
+
+        class AssertNumOutputBackend:
+            """
+            A backend that checks the number of output for compiled graph, and
+            return the graph as is.
+            """
+
+            def __init__(self, expected_num_output: int):
+                self.expected_num_output = expected_num_output
+
+            def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+                outputs = gm(*example_inputs)
+                test_case_self.assertEqual(self.expected_num_output, len(outputs))
+                return gm
+
+        @torch.compile(backend=AssertNumOutputBackend(2))
+        def test_returning_nested_func_with_captured_tensor():
+            x = torch.rand(1)
+
+            def func():
+                return x + x
+
+            # Returning `func` forces dynamo to output `x` in the compiled graph, so
+            # that we can store it as `func`'s closure.
+            return func, func()
```

Comment:
the outputs are (x, x + x) ?

Reply:
Oh I meant, what are the 2 outputs in the compiled graph?

Reply:
Ah, yep. Lemme document that.